Tests/NIOHTTP1Tests/HTTPResponseCompressorTest.swift

```diff
@@ -39,11 +39,11 @@ private class PromiseOrderer {
         promise.futureResult.whenComplete {
             let priorFutures = self.promiseArray[0..<thisPromiseIndex]
             let subsequentFutures = self.promiseArray[(thisPromiseIndex + 1)...]
-            let allPriorFuturesFired = priorFutures.map { $0.futureResult.isFulfilled }.reduce(true, { $0 && $1 })
-            let allSubsequentFuturesUnfired = subsequentFutures.map { $0.futureResult.isFulfilled }.reduce(false, { $0 || $1 })
+            let allPriorFuturesFired = priorFutures.map { $0.futureResult.isFulfilled }.allSatisfy { $0 }
```

Comment:
I think all of these use cases would be clearer, and probably more performant, without the initial map:

```suggestion
            let allPriorFuturesFired = priorFutures.allSatisfy { $0.futureResult.isFulfilled }
```